src/fiatExchanges/CoinbasePaymentSection.test.tsx

```diff
@@ -0,0 +1,81 @@
+import { render } from '@testing-library/react-native'
+import * as React from 'react'
+import { Provider } from 'react-redux'
+import { MockStoreEnhanced } from 'redux-mock-store'
+import {
+  CoinbasePaymentSection,
+  CoinbasePaymentSectionProps,
+} from 'src/fiatExchanges/CoinbasePaymentSection'
+import { PaymentMethod } from 'src/fiatExchanges/utils'
+import { createMockStore } from 'test/utils'
+import { mockProviders } from 'test/values'
+
+describe('CoinbasePaymentSection', () => {
+  let props: CoinbasePaymentSectionProps
+  let mockStore: MockStoreEnhanced
+  beforeEach(() => {
+    props = {
+      coinbaseProvider: mockProviders.find(
+        (quote) => quote.paymentMethods[0] === PaymentMethod.Coinbase
+      )!,
+      setNoPaymentMethods: jest.fn(),
+    }
+    mockStore = createMockStore({
+      app: {
+        coinbasePayEnabled: false,
+      },
+    })
+  })
+  it('shows nothing if coinbase is restricted and feature flag is false', async () => {
+    props.coinbaseProvider!.restricted = true
+    const { queryByText } = render(
+      <Provider store={mockStore}>
+        <CoinbasePaymentSection {...props} />
+      </Provider>
+    )
+    expect(queryByText('Coinbase Pay')).toBeFalsy()
+    expect(props.setNoPaymentMethods).toHaveBeenCalled()
```

Comment:
Jacob suggested you just take the `setNoPaymentMethods` part out of the component, which is probably right. But if you didn't, you'd want to test what it was called with, not just that it was called